plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFieldsFrontend.php

```diff
@@ -186,42 +186,61 @@ public function edit_account_form_fields() {
 	 * @param integer $user_id User ID.
 	 */
 	public function save_account_form_fields( $user_id ) {
-		// phpcs:disable WordPress.Security.NonceVerification.Missing
-		$customer          = new WC_Customer( $user_id );
-		$additional_fields = $this->checkout_fields_controller->get_fields_for_location( 'contact' );
-		$field_values      = array();
 
-		foreach ( array_keys( $additional_fields ) as $key ) {
-			$post_key = CheckoutFields::get_group_key( 'other' ) . $key;
-			if ( ! isset( $_POST[ $post_key ] ) ) {
-				continue;
-			}
+		try {
+			$customer = new WC_Customer( $user_id );
 
-			$field_value = $this->checkout_fields_controller->sanitize_field( $key, wc_clean( wp_unslash( $_POST[ $post_key ] ) ) );
-			$validation  = $this->checkout_fields_controller->validate_field( $key, $field_value );
+			$additional_fields = $this->checkout_fields_controller->get_fields_for_location( 'contact' );
 
-			if ( is_wp_error( $validation ) && $validation->has_errors() ) {
-				wc_add_notice( $validation->get_error_message(), 'error' );
-				continue;
+			$field_values = array();
+
+			foreach ( $additional_fields as $key => $field_data ) {
+				// We can't skip, field might be required.
+				$field_value = trim( wc_clean( wp_unslash( $_POST[ $key ] ?? '' ) ) ); // phpcs:ignore WordPress.Security.NonceVerification.Missing
```

Comment:
`wc_clean` uses `sanitize_text_field`. The later already trims for us. This is not necccessary.

Reply:
`$_POST[ $key ]` is wrong. It doesn't match the form field because the form field `name` is prefixed with the `group`. e.g. `CheckoutFields::get_group_key( $group ) . $key`